src/ORM/FieldType/DBComposite.php

```diff
@@ -29,26 +30,21 @@ abstract class DBComposite extends DBField
      * holds an array of composite field names.
      * Don't include the fields "main name",
      * it will be prefixed in {@link requireField()}.
-     *
-     * @config
-     * @var array
      */
-    private static $composite_db = [];
+    private static array $composite_db = [];
 
     /**
      * Marker as to whether this record has changed
      * Only used when deference to the parent object isn't possible
      */
-    protected $isChanged = false;
+    protected bool $isChanged = false;
 
     /**
      * Either the parent dataobject link, or a record of saved values for each field
-     *
-     * @var array|DataObject
      */
-    protected $record = [];
+    protected array|ViewableData $record = [];
```

Comment:
Since this wasn't strongly typed before, I'm hesitant to jump straight to typing it to `DataObject`, since we don't really know how these are being used in the wild.
I'm cautiously typing against `ViewableData` instead. I've checked and everywhere that I've gone for `ViewableData` instead of `DataObject` is safe in terms of what methods are being called.